gulpfile.js

```diff
@@ -34,17 +19,35 @@ gulp.task('edit', function() {
   });
 });
 
-gulp.task('build', function (cb) {
+gulp.task('build', (cb) => {
   exec('npm run build', function (err, stdout, stderr) {
     console.log(stderr);
     cb(err);
   });
 });
 
-gulp.task('reload', ['build'], function () {
+gulp.task('reload', gulp.series('build'), () => {
   gulp.src(DIST_DIR).pipe(gulpConnect.reload())
 });
 
-gulp.task('watch', function () {
-  gulp.watch(['spec/**/*', 'web/**/*'], ['reload']);
+gulp.task('watch', () => {
+  gulp.watch(['spec/**/*', 'web/**/*'], gulp.series('reload'));
+});
+
+gulp.task('ui', () => {
+  portfinder.getPort({port: 3000}, function (err, port) {
+    return gulpConnect.server({
+      root: [DIST_DIR],
+      livereload: true,
+      port: port,
+      middleware: (gulpConnect, opt) => {
+        return [
+          cors()
+        ]
+      }
+    });
+  });
+});
+
+gulp.task('serve', gulp.parallel('ui', 'build', 'watch', 'edit'), () => {
```

Comment:
I'm not very familiar with using gulp for building node.js workflows, but I guess this is newer/cleaner way to define things? Seems intuitive enough 🪄